wayshot/src/wayshot.rs

```diff
@@ -173,32 +228,82 @@ fn main() -> Result<(), Box<dyn Error>> {
     if extension != libwayshot::EncodingFormat::Png {
         log::debug!("Using custom extension: {:#?}", extension);
     }
+    if frame_copy.0.len() == 1 {
+        let frame_copy = &frame_copy.0[0];
+        if args.is_present("stdout") {
+            let stdout = stdout();
+            let mut writer = BufWriter::new(stdout.lock());
+            libwayshot::write_to_file(&mut writer, extension, frame_copy)?;
+        } else {
+            let path = if args.is_present("file") {
+                args.value_of("file").unwrap().trim().to_string()
+            } else {
+                let time = match SystemTime::now().duration_since(UNIX_EPOCH) {
+                    Ok(n) => n.as_secs().to_string(),
+                    Err(_) => {
+                        log::error!("SystemTime before UNIX EPOCH!");
+                        exit(1);
+                    }
+                };
 
-    if args.is_present("stdout") {
-        let stdout = stdout();
-        let writer = BufWriter::new(stdout.lock());
-        libwayshot::write_to_file(writer, extension, frame_copy)?;
+                time + match extension {
+                    libwayshot::EncodingFormat::Png => "-wayshot.png",
+                    libwayshot::EncodingFormat::Jpg => "-wayshot.jpg",
+                    libwayshot::EncodingFormat::Ppm => "-wayshot.ppm",
+                }
+            };
+
+            libwayshot::write_to_file(&mut File::create(path)?, extension, frame_copy)?;
+        }
     } else {
-        let path = if args.is_present("file") {
-            args.value_of("file").unwrap().trim().to_string()
+        let mut images = Vec::new();
+        let (frame_copy, region) = frame_copy;
+        let (width, height) = region.unwrap();
+        for frame_copy in frame_copy {
+            let mut buff = Cursor::new(Vec::new());
+            libwayshot::write_to_file(&mut buff, extension, &frame_copy)?;
+            let image = image::load_from_memory(buff.get_ref())?;
+            let image = image::imageops::resize(
+                &image,
+                width as u32,
+                height as u32,
+                image::imageops::FilterType::Gaussian,
+            );
+            images.push(image);
+        }
+        use image::imageops::overlay;
+        let mut image_bottom = images[0].clone();
+        for image in images {
+            overlay(&mut image_bottom, &image, 0, 0);
+        }
+        if args.is_present("stdout") {
+            let stdout = stdout();
+            let mut buff = Cursor::new(Vec::new());
+
+            let mut writer = BufWriter::new(stdout.lock());
+            image_bottom.write_to(&mut buff, image::ImageFormat::Png)?;
```

Comment:
I think I didn't notice this line, but should we not take the users defined extension format into consideration here instead of hardcoding png or am I mistaken? @Decodetalkers

Reply:
sorry, I have missed that